File: src/Test/AckermannFFI.go
package Test_AckermannFFI


func ack(m, n int) int {
	if m == 0 {
		return n + 1
	}
	if m > 0 && n == 0 {
		return ack(m-1, 1)
	}
	return ack(m-1, ack(m, n-1))
}

func RunAckermannFFI(limit float64) float64 {
	return float64(ack(3, 4))
}

File: src/Test/AckermannFFI.erl
-module(test_ackermannFFI@foreign).
-export([runAckermannFFI/1]).
runAckermannFFI(_Dummy) -> ack(3, 4).
ack(0, N) -> N + 1;
ack(M, 0) when M > 0 -> ack(M - 1, 1);
ack(M, N) -> ack(M - 1, ack(M, N - 1)).
